server/controllers/notification.js

```diff
@@ -0,0 +1,95 @@
+const asyncHandler = require("express-async-handler");
+const Notification = require("../models/Notification");
+const User = require("../models/User");
+
+exports.createNotification = asyncHandler(async (req, res) => {
+  const date = new Date();
+  const {title, type, senderId, recipientId, message} = req.body;
+  await User.findOne({_id: senderId}, function(err) {
+    if(err) {
+        res.status(404).send("Sender not found!!")
+    }
+  });
+  await User.findOne({_id: recipientId }, function(err) {
+    if(err) {
+        res.status(404).send("Receiver not found");
+    }
+  });
+
+  try {
+    const notification = await Notification.create({
+      type,
+      title,
+      recipientId,
+      senderId,
+      message,
+      date
+    });
+    res.status(201).json(notification);
+  }
+
+  catch (err) {
+  res.status(500).json(err);
+    }
+  }
+);
+
+exports.getAllNotifications = asyncHandler(async (req, res) => {
+  const {recipientId} = req.body;
+  await User.findOne({_id: recipientId }, function(err) {
+    if(err) {
+        res.status(404).send("Receiver not found!");
+    }
+  });
+  try {
+    const allNotifications = await Notification.find({
+      _id: recipientId,
+    });
+    if (!allNotifications) {
+      res.send("No notifications found!")
+    }
+    res.status(200).json(allNotifications);
+  } catch (err) {
+    res.status(500).json(err);
+  }
+});
+
+exports.markNotificationRead = asyncHandler(async (req, res, next) => {
+  const notification_id = req.params.notification_id;
+  const notification = Notification.findOne({_id: notification_id}, function(err) {
+    if(err) {
+        res.status(404).send("Notification not found!!");
+    }
+  })
+  if(notification) {
+    const notification = await Notification.updateOne({ _id: notification_id }, { isRead: true }, function(err) {
+        if(err) {
+            res.status(500).send("Internal Server Error!");
+        }
+
+        res.status(200).json(notification);
+    })
+  }
+});
+
+
+exports.getUnreadNotifications = asyncHandler(async (req, res, next) => {
+  const {recipientId} = req.body;
```

Comment:
GET requests don't have a request body, maybe consider using the logged in user's id (`req.user.id`) as he is the intended recipient for those notifications?

Ref: https://dropbox.tech/developers/limitations-of-the-get-method-in-http

Reply:
Read the document, thanks Rajiv, fixed